connect/cli/plugins/shared/translations_synchronizers.py

```diff
@@ -0,0 +1,45 @@
+import click
+
+from connect.cli.plugins.shared.translation_sync import TranslationsSynchronizer
+from connect.cli.plugins.shared.translation_attr_sync import TranslationAttributesSynchronizer
+from connect.cli.plugins.shared.utils import (
+    get_translation_attributes_sheets,
+    wait_for_autotranslation,
+)
+from connect.cli.plugins.shared.exceptions import SheetNotFoundError
+
+
+def translations_sync(client, config, input_file, stats, save):
+    synchronizer = TranslationsSynchronizer(client, config.silent, stats)
+    synchronizer.open(input_file, 'Translations')
+    synchronizer.sync()
+    if save:
+        synchronizer.save(input_file)
+    return synchronizer
+
+
+def translation_attributes_sync(worksheet, translation_id, client, config, input_file, stats, save, new_translation):
+    synchronizer = TranslationAttributesSynchronizer(client, config.silent, stats, new_translation)
+    synchronizer.open(input_file, worksheet)
+    synchronizer.sync(translation_id)
+    if save:
+        synchronizer.save(input_file)
+
+
+def sync_product_translations(client, config, input_file, stats, save=True):
+    try:
+        synchronizer = translations_sync(client, config, input_file, stats, save)
+        translations_autotranslating = synchronizer.translations_autotranslating
+        new_translations = synchronizer.new_translations
+    except SheetNotFoundError as e:
+        if not config.silent:
+            click.secho(str(e), fg='blue')
+    new_translations.insert(0, None)
+    for new_translation, sheetname in zip(new_translations, get_translation_attributes_sheets(input_file)):
```

Comment:
is this assuming that there will be the same number of new translations as translation sheets? I think that this is a particular case when cloning a product, but when syncing this might not be so.

Reply:
This is updated. Now we have a translation that goes inside the `TranslationAttributesSynchronizer` and it never knows if this is a cloned translation or a sync translation, and also use the same ` _update_attributes` method, so we save duplicate code while discard unexpected behavior from `sync` command.